### ISSUE TITLE
[Feature]: Post file directly from frontend while using File Picker Widget

### ISSUE THREAD

**Issue by JhouXerox** [appsmithorg/appsmith#17045]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Post file directly from frontend while using File Picker Widget,without pass it to backend,and let backend execute.

### Why should this be worked on?

We've a solution that backend side will sign a token to frontend when any user want to upload something, but the token is only 60 seconds, after that, this token can't upload anymore.
I was using this widget to upload a little large file about 60Mb,I found it will cost about 3min.
I test in my local, the action is cost about 90 seconds.

 my action of filepicker.onfileselection is 

return await Sign.run()
		 .then(async function(){
			 await UploadFile.run()
			 showAlert("upload success","success")
		 }).catch(function(){
			 showAlert(UploadFile.data + JSON.stringify(UploadFile.responseMeta),"error")
		 })

it's always return token is expired.
maybe it's because frontend side pass the file to backend take a lot of time, so that token is already expired when the backend execute the UploadFile.
I'm trying to make a simple site to post file from frontend side directly, and use iframe widget to embed the appsmith.

can we get an official option to control? or backend can raise some trigger on backend has got the file from frontend?

**Comment by Nikhil-Nandagopal** [comment]
@arunvjn @bharath31 will this be possible after enabling fetch?

**Comment by arunvjn** [comment]
@Nikhil-Nandagopal  Yes, we should able to do the file upload directly using fetch when we enable it.

**Comment by JhouXerox** [comment]
@arunvjn 
I'm tying to use fetch api to upload file,there's some questions.
1. I try to upload file directly like the pic. But it send string"[object Object]" in file field.
![image](https://user-images.githubusercontent.com/24403596/206958312-6c157893-51d1-4f94-9722-7eefcb1be2ce.png)

2. I try to convert file to File type, but I found that file size is changed.
![image](https://user-images.githubusercontent.com/24403596/206958451-1fe5dd42-766b-4014-b388-226c6468a5b8.png)
<img width="591" alt="image" src="https://user-images.githubusercontent.com/24403596/206958650-7fc67543-33ce-4541-9a4a-d973467c04cd.png">

need your help

**Comment by JhouXerox** [comment]
#18861 https://github.com/appsmithorg/appsmith/issues/18861

**Comment by arunvjn** [comment]
@JhouXerox Could you try converting File.data(base64/binary) into a blob and create a file object out of the blob ?

**Comment by JhouXerox** [comment]
size is not same. And converted file can't unzip. It's broken.

**Comment by JhouXerox** [comment]
size of original file in FilePicker is 27853.
convert to blob by "var blob = new Blob([file.data],{type:file.type})", blob.size is 39258
convert to File by "var f = new File([blob],file.name,{type:file.type});" File.size is 39258.

**Comment by JhouXerox** [comment]
need to add:
var buffer = Uint8Array.from(file.data, c => c.charCodeAt(0))

![image](https://user-images.githubusercontent.com/24403596/207001382-bc674169-a5b3-4be0-b037-76e1fc46aeb6.png)

It's correct.
now it can be unzipped.

should it add to the fetch api internal to convert file.data?

**Comment by JhouXerox** [comment]
maybe it can be provided as a util class or something to help others to convert file.data

**Comment by JhouXerox** [comment]
another question is appeared.
buffer.size's max is 10w.
when file.data over 10w ,buffer can only take 10w.

**Comment by arunvjn** [comment]
@JhouXerox Glad that it is working now. 
Kindly post the rest of your questions in the support channel on discord. We would be happy to help you.

**Comment by JhouXerox** [comment]
@arunvjn 
my appsmith's version is now 1.8.12, and my old page is built from 1.8.0,for now in it's filepicker, file.data,length can't over 10w,but it's can over 10w in new page.
how should I deal with it?

**Comment by JhouXerox** [comment]
@arunvjn 
and file.data will convert to an blob info in sometimes.
it can't handle in fetch
![image](https://user-images.githubusercontent.com/24403596/207026209-829ed0d8-c41d-4d3b-ae06-f14c9f176daf.png)

**Comment by satbir121** [comment]
@ravikp7 Can you please look into uploading large files using fetch? Post a code snippet here.

**Comment by JhouXerox** [comment]
Is v1.52 allowed to upload file from foreground directly to S3 ? in 1.8 it has to proxied by appsmith backend.
and is there anyware allow to upload big file over 200M?
